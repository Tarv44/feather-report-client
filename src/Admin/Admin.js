import React, {Component} from 'react';
import styles from './Admin.module.css';
import ProductContext from '../productContext';
import EditProduct from '../EditProduct/EditProduct';
import ProductsSection from '../ProductsSection/ProductsSection';
import { NavLink } from 'react-router-dom';
import config from '../config';

const { API_ENDPOINT } = config

export default class Admin extends Component {
    static contextType = ProductContext;

    state = {
        company: {},
        categories: [],
        products: []
    }

    componentDidMount() {
        const co_path = this.context.company.pathname

        const options = {
            method: 'GET',
            headers: {
                'content-type': 'application/json'
            }
        }

        if (co_path !== undefined) {
            fetch(`${API_ENDPOINT}/companies/${co_path}/products`,options)
                .then(res => {
                    if (!res.ok) {
                        return res.json().then(err => { throw err })
                    }
                    return res.json()
                })
                .then(res => {
                    const { company, categories, products } = res
                    this.setState({ company, categories, products})
                })
        }
    }

    addProduct = (product) => {
        const products = this.state.products
        products.unshift(product)
        this.setState(products)
    }

    updateProduct = (product) => {
        const products = this.state.products
        const i = products.findIndex(p => p.id === product.id)
        products[i] = product
        this.setState({ products })
    }

    updateCats = (cat) => {
        const categories = this.state.categories
        categories.push(cat)
        this.setState({ categories })
    }

    render() {
        return this.context.company.title 
        ? (
            <>
                <header className={styles.header}>
                    <h1>{this.context.company.title}</h1>
                    <h3>Admin Page</h3>
                    <NavLink 
                        className={styles.customer} 
                        to={`/co/${this.context.company.pathname}/products`}
                    >
                        Customer Page
                    </NavLink>
                </header>
                <main className={styles.main}>
                    <section className={styles.edit}>
                        <EditProduct
                            allCats={this.state.categories} 
                            addProduct={this.addProduct}
                            handleCats={this.updateCats}
                        />
                    </section>
                    <ProductsSection
                        products={this.state.products}
                        updateProduct={this.updateProduct}
                        allCats={this.state.categories}
                        handleCats={this.updateCats}
                        type='edit'
                    />
                </main>
            </>
        )
        : (
            <main>
                <h2>Log in to use Admin page.</h2>
                <NavLink to={'/login'}>Log In</NavLink>
            </main>
        )
    }
}