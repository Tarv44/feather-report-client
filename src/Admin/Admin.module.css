.header {
    text-align: center;
    min-width: 100%;
}

.main {
    min-width: 100%;
}

.customer {
    border: solid #26547C 1px;
    padding: .25em 1em;
    color: #26547C;
}