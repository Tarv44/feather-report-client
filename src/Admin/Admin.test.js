import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import Admin from './Admin'
import { BrowserRouter } from 'react-router-dom'

describe('----- Admin -----', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(
            <BrowserRouter>
                <Admin/>
            </BrowserRouter>
        , div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(
                <BrowserRouter>
                    <Admin/>
                </BrowserRouter>
            )
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})

