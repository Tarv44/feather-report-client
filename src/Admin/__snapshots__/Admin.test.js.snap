// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- Admin ----- renders the UI as expected 1`] = `
<main>
  <h2>
    Log in to use Admin page.
  </h2>
  <a
    aria-current={null}
    href="/login"
    onClick={[Function]}
  >
    Log In
  </a>
</main>
`;
